src/sagemaker/tensorflow/estimator.py

```diff
@@ -177,6 +177,11 @@ def __init__(
                 fw.python_deprecation_warning(self._framework_name, defaults.LATEST_PY2_VERSION)
             )
         self.framework_version = framework_version
+        if self.framework_version and "2.16" in self.framework_version:
+            self.override_fw_version = "2.16"
```

Comment:
```
To handle this, in Tensorflow Estimator, we just use the major.minor version for TF 2.16.* images when retrieving the image_uri.
```

Does this force the estimator itself to default framework version to the short tag? If so, customers pinned to a specific patch version of TF will presumably be force upgraded, and this will be unexpected behavior for seasoned pysdk users. Can't we accomplish testing parity without adding version specific handlers in the source code?

Reply:
Yes, had the same opinion about customers getting auto-transferred to the latest version if we change the source code. Modified the testing logic to appropriately handle this case. Thanks!